### PR TITLE
Worker Prometheus Stats

### PR DESCRIPTION
**Pull request by dccurtis**
Address #396 

Adds a Prometheus metrics endpoint to the mass worker (http://worker_host:9999/metrics) which can be scraped by Prometheus.

Updates to docker-compose are done to allow for local development when using prometheus and Grafana.  Note: Configuring the Grafana datasource through the config file is still a work in progress.

The following counters are part of this PR to demonstrate that the data is available in Prometheus.  We will surely be adding to this:
```
get_report_files_attempts_count
report_file_download_count
report_file_download_error_count
process_report_attempts_count
process_report_error_count
report_summary_attempts_count
charge_update_attempts_count
```

I fully expect that we will revamp how the stats are setup in `prometheus_stats.py`, the main purpose of this PR is to get the infrastructure in place and show some examples of how it's used.

Example of the worker endpoint:
```
# HELP charge_update_attempts_count_total Multiprocess metric
# TYPE charge_update_attempts_count_total counter
charge_update_attempts_count_total 12.0
# HELP get_report_files_attempts_count_total Multiprocess metric
# TYPE get_report_files_attempts_count_total counter
get_report_files_attempts_count_total{provider_type="OCP"} 3.0
# HELP process_report_attempts_count_total Multiprocess metric
# TYPE process_report_attempts_count_total counter
process_report_attempts_count_total{provider_type="OCP"} 12.0
# HELP report_summary_attempts_count_total Multiprocess metric
# TYPE report_summary_attempts_count_total counter
report_summary_attempts_count_total{provider_type="OCP"} 12.0
# HELP get_report_files_attempts_count_total Number of ingest attempts
# TYPE get_report_files_attempts_count_total counter
# HELP report_file_download_error_count_total Number of report file download errors
# TYPE report_file_download_error_count_total counter
# HELP process_report_attempts_count_total Number of report files attempted processing
# TYPE process_report_attempts_count_total counter
# HELP process_report_error_count_total Number of report files attempted processing
# TYPE process_report_error_count_total counter
# HELP report_summary_attempts_count_total Number of report summary attempts
# TYPE report_summary_attempts_count_total counter
# HELP charge_update_attempts_count_total Number of derivied cost update attempts
# TYPE charge_update_attempts_count_total counter
charge_update_attempts_count_total 0.0
# TYPE charge_update_attempts_count_created gauge
charge_update_attempts_count_created 1.5542205837949038e+09
```

Visible in Prometheus
prometheus_example

Scaling up the workers, prometheus sees each worker pod
image
image

### CODE REVIEW

File: openshift/worker.yaml [hunk 1]
```diff
@@ -14,6 +14,25 @@ labels:
   app: masu
   template: masu-template
 objects:
+- kind: Service
```
Comment (chambridge) [inline openshift/worker.yaml]: Might be nicer to pull this out into its own yaml if that we could replace the `${NAME}-worker-0` if we scale up, unless there is another way to dynamically create the service.

**Comment by dccurtis** [comment]
Yes, that is correct.  The only purpose of the stuff in the `docker_resources` directory is for local development.

**Comment by blentz** [comment]
Yeah, I just double-checked myself after I wrote that. I overlooked the docker-compose. Mea Culpa.

**Comment by dccurtis** [comment]
BTW I'll probably not merge this until I address @chambridge comments. @chambridge if you want to pull this early feel free to merge away and I'll address your comments in a separate PR.

**Pull request merged**